ivy/backend_handler.py

```diff
@@ -98,6 +98,80 @@ def _determine_backend_from_args(args):
                 return importlib.import_module(module_name)
 
 
+def fn_name_from_version_specific_fn_name(name, version):
+    """
+
+    Parameters
+    ----------
+    name
+        the version specific name of the function for which the version support
+        is to be provided.
+    version
+        the version of the current framework for which the support is to be
+        provided, the version is inferred by importing the framework
+    Returns
+    -------
+        the name of the original function which will then point to the version
+        specific function
+
+    """
+    version = str(version)
+    if version.find("+") != -1:
+        version = int(version[: version.index("+")].replace(".", ""))
+    else:
+        version = int(version.replace(".", ""))
+    if "_to_" in name:
+        i = name.index("_v_")
+        e = name.index("_to_")
+        version_start = name[i + 3 : e]
+        version_start = int(version_start.replace("p", ""))
+        version_end = name[e + 4 :]
+        version_end = int(version_end.replace("p", ""))
+        if version in range(version_start, version_end + 1):
+            return name[0:i]
+    elif "_and_above" in name:
+        i = name.index("_v_")
+        e = name.index("_and_")
+        version_start = name[i + 3 : e]
+        version_start = int(version_start.replace("p", ""))
+        if version >= version_start:
+            return name[0:i]
+    else:
+        i = name.index("_v_")
+        e = name.index("_and_")
+        version_start = name[i + 3 : e]
+        version_start = int(version_start.replace("p", ""))
+        if version <= version_start:
+            return name[0:i]
+
+
+def set_backend_to_specific_version(backend):
+    """
+
+    Parameters
+    ----------
+    backend
+        the backend module for which we provide the version support
+    Returns
+        The function doesn't return anything and updates the backend __dict__
+        to make the original function name to point to the version specific one
+
+    -------
+
+    """
+    f = str(backend.__name__)
+    f = f[f.index("backends") + 9 :]
+
+    f = importlib.import_module(f)
```

Comment:
is this not exactly what `backend` already is? why do we need to import it again?

Reply:
Actually, the variable backend here would be a module say `ivy.functional.backends.torch`. We want the `torch` part so that we can import the original `torch` from the user's system and get its version.